pilot/pkg/xds/rds.go

```diff
@@ -42,27 +43,14 @@ var skippedRdsConfigs = sets.New[kind.Kind](
 )
 
 func rdsNeedsPush(req *model.PushRequest, proxy *model.Proxy) bool {
-	if proxy.Type == model.Ztunnel {
-		// Not supported for ztunnel
-		return false
-	}
-	if req == nil {
-		return true
-	}
-	if !req.Full {
-		// RDS only handles full push
-		return false
-	}
-	// If none set, we will always push
-	if len(req.ConfigsUpdated) == 0 {
-		return true
-	}
-	for config := range req.ConfigsUpdated {
-		if !skippedRdsConfigs.Contains(config.Kind) {
-			return true
+	return xdsNeedsPush(req, proxy, xds_model.RouteType, func(req *model.PushRequest, proxy *model.Proxy) bool {
+		for config := range req.ConfigsUpdated {
```

Comment:
Do we need a callback like this? why not just something like


```go
if res, ok := commonChecks(); ok { return res }
// type specific
for config := range req.ConfigsUpdated {
			if !skippedRdsConfigs.Contains(config.Kind) {
				return true
			}
}
return false
```

?

Reply:
> if res, ok := commonChecks(); ok { return res }

Yes. This definitely works and this was my first implementation. Later changed to call back. If this is more preferred, I can change it back